src/components/todoList/TodoList.js

```diff
@@ -0,0 +1,121 @@
+import { Component } from 'react';
+import './todoList.scss';
+
+export class TodoList extends Component {
+  state = {
+    todos: [],
+    initialTodos: [],
+    filterValue: ''
+  }
+
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/todos')
+      .then(response => response.json())
+      .then(todos => this.setState({
+        todos: todos.slice(0, 20),
+        initialTodos: todos.slice(0, 20)
+      }));
+  }
+
+  deleteTodo = (id) => {
+    const { todos } = this.state;
+    this.setState({
+      todos: todos.filter(todo => todo.id !== id),
+      initialTodos: todos.filter(todo => todo.id !== id)
+    });
+  }
+
+  completeTodo = (todoId) => {
+    const { todos } = this.state;
+    const item = todos.find(item => item.id === todoId);
+    this.setState({
+      todos: [
+        ...todos.slice(0, todoId - 1),
+        { ...item, completed: true, inprogress: false },
+        ...todos.slice(todoId)]
+    });
```

Comment:
looks no good. We can't calculate index of todo in array relying on its `id`. 
it can be done simplier:
```
const item = todos.find(item => item.id === todoId);

item.completed = true;
this.setState({ todos: [...todos] })
```